notifications.bs

```diff
@@ -539,14 +555,48 @@ interpreted as a language tag. Validity or well-formedness are not enforced. [[!
 platform supports activation, the user agent must (unless otherwise specified) run these steps:
 
 <ol>
+ <li><p>Let <var>action</var> be null.
+
+ <li><p>If one of <var>notification</var>'s <a for=notification>actions</a> was activated by the end
+ user, then set <var>action</var> to that <a for=/>action</a>.
+
+ <li><p>Let <var>navigationURL</var> be <var>notification</var>'s
+ <a for=notification>navigation URL</a>.
+
+ <li>
+  <p>If <var>action</var> is non-null, then set <var>navigationURL</var> to <var>action</var>'s
+  <a for=action>navigation URL</a>.
+
+  <p class=note>This intentionally makes it so that when an <a for=/>action</a>'s
+  <a for=action>navigation URL</a> is null, it falls through to the <code>click</code> event,
+  providing more flexibility to the web developer.
+
+ <li>
+  <p>If <var>navigationURL</var> is non-null:
+
+  <ol>
+   <li><p>If there is an existing <a for=/>top-level traversable</a> <var>traversable</var> within
+   the user agent's <a for="user agent">top-level traversable set</a> whose
+   <a for="navigable">active document</a>'s <a for=Document>URL</a> <a for=url>equals</a>
+   <var>navigationURL</var> with <a for=url/equals><i>exclude fragments</i></a> set to true, then
+   <a for=/>navigate</a> <var>traversable</var> to <var>navigationURL</var> and return. If there are
+   multiple <a for=/>top-level traversables</a> that satisfy the condition the user agent has to
+   pick one in an <a>implementation-defined</a> manner.
```

Comment:
Given "losing the data by accidental navigation" is an existing problem with existing solution, Mozilla is against letting this fully implementation defined. We prefer to giving some basic heuristic to prevent it, unless there's a proof that selecting random traversable will not cause a problem.

Here's our suggested revision:

1. Let |traversables| be the user agent’s top-level traversable set whose active document’s URL equals |navigationURL| with exclude fragments set to true. (This part is copied from the PR, but see the above comment)
2. Let |selected| be null.
3. For each |traversable| in |traversables|,
   1. If |traversable| needs beforeunload, continue.
   2. Let |activeDocument| be |traversable|'s active document.
   3. Let |focusState| be the result of running the [has focus steps] with |activeDocument| as the argument.
   4. If |focusState| is true, let |selected| be |traversable| and break.
   5. If |activeDocument|'s visibility state is "visible", let |selected| be |traversable| and break.
4. If |selected| is null and |traversables| is not empty, let |selected| be the first item of |traversables|.
5. If |selected| is null,
   1. Create a fresh top-level traversable given |navigationURL| and return.
6. Navigate |selected| to |navigationURL|.
7. Run the focusing steps with |selected|.

Reply:
If the website is not robust against navigation, wouldn't they simply mint a new URL?